src/controllers/PipServerController.ts

```diff
@@ -127,10 +127,50 @@ export class PipServerController implements Disposable {
     return this._checkPipInstallPromise;
   };
 
-  getAvailablePorts = (): Port[] => {
-    return [...this._portRange]
-      .filter(port => !this._serverUrlTerminalMap.has(port))
-      .sort();
+  /**
+   * Gets the next available port for starting a pip server.
+   * @returns A port number or `null` if no ports are available.
+   */
+  getNextAvailablePort = (): Port | null => {
+    for (let i = 10000; i < 10050; ++i) {
```

Comment:
Why stop at 10050? Why not like 10999?

Reply:
It was arbitrary. Figured 50 was even higher than system resources would probably allow